src/main/java/javaserver/ResponseBuilder.java

```diff
@@ -24,19 +24,34 @@ private void setResponseData(String statusLine, String header, String body) {
 
 	private String getStatusLine() {
 		HTTPStatusCode responseCode;
-		if (isSupported(request)) {
-			responseCode = HTTPStatusCode.TWO_HUNDRED;
+		if (requestIsSupported()) {
+			if (RequestParser.isRedirect(request)) {
+				responseCode = HTTPStatusCode.THREE_OH_TWO;
+			} else {
+				responseCode = HTTPStatusCode.TWO_HUNDRED;
+			}
+		} else if (isNotAllowed(request)) {
+			responseCode = HTTPStatusCode.FOUR_OH_FIVE;
 		} else {
 			responseCode = HTTPStatusCode.FOUR_OH_FOUR;
 		}
 		return responseCode.getStatusLine();
 	}
-	
-	private boolean isSupported(Request request) {
+
+	private boolean requestIsSupported() {
 		String requestType = request.getMethod();
-		if (request.routeOptions() != null && (Arrays.asList(request.routeOptions()).contains(requestType))) {
+		if (getRouteOptions() != null && (Arrays.asList(getRouteOptions()).contains(requestType))) {
 			return true;
-		} else if (request.hasVariableParams()) {
+		} else if (RequestParser.hasVariableParams(request)) {
```

Comment:
This conditional doesn't seem correct - it will return true for any request that contains `?` regardless of whether or not the route is valid.